### ISSUE TITLE
Will not run under Linux

### ISSUE THREAD

**Issue by hatesigningup** [ToniMacaroni/RedManager#13]
Sons of the Forest runs fantastic under Wine for Linux, but I can't for the life of me get RedModManager to run.

I know this isn't as bug, but was hoping you could give me some clue as to how the programme works so I have a better chance of getting it running.

I installed RedMod and all my mods in my Windows copy of Sons and copied the entire folder over to Linux.  It runs, but there is no sign of RedMod.   As far as I can tell there are no other external hidden folders that RedMod needs.

I don't understand how RedMod gets started, it seems that everything is in the Sons folder and it somehow side-loads.  The question is how?

Any ideas?

Thanks and regards.

**Comment by hatesigningup** [comment]
OK, so after a lot of head scratching and researching, I think I finally see how this whole thing works.

1. The game engine (Unity) on startup checks the Game's root folder for any extra .dll files.
2. It finds dobby.dll & version.dll that point to the sideloading code (this is the RedLoader part of the RedModManager package).
3. One of those .dlls redirects execution to the recompiler (Il2cpp) which modifies the code to allow the injection of RedModManager.
4. Execution then returns to the "normal" game code.

So the problem lies somewhere in this loading chain (not RedModManager), and sadly there are no logs to follow which might point me to the culprit.

This is just my best guess based on my research, but if anyone has any more info I would be grateful for any leads.

Regards.

it seems that the problem lies with dobby.dll.  Turns out that

**Comment by Dankeatermidir** [comment]
Hi, I have solution for this:
1. Follow manual installation
2. Run **protontricks --gui**
3. Choose Sons of the Forest -> default prefix -> winecfg
4. In libraries tab add **version.dll** and apply

It's really similar to running BepInEx under proton (https://docs.bepinex.dev/articles/advanced/proton_wine.html)
I've also installed dotnet6 using protontricks, though I'm not sure if it had any effect.

EDIT:
Just realized It's RedManager not RedLoader, hope it helps anyway.

**Comment by hatesigningup** [comment]
> Hi, I have solution for this:
> 
>     1. Follow manual installation
> 
>     2. Run **protontricks --gui**
> 
>     3. Choose Sons of the Forest -> default prefix -> winecfg
> 
>     4. In libraries tab add **version.dll** and apply
> 
> 
> It's really similar to running BepInEx under proton (https://docs.bepinex.dev/articles/advanced/proton_wine.html) I've also installed dotnet6 using protontricks, though I'm not sure if it had any effect.
> 
> EDIT: Just realized It's RedManager not RedLoader, hope it helps anyway.

SOLVED - it worked!!!

I first ran the command you gave "protontricks --gui", it complained about 64bit and then about WOW or something but I simply ignored ALL error messages and followed the rest of your instructions.  Added "version", saved and exited.

The Linux version of Sons wouldn't even load anymore, so I rebooted to Windows, updated RedModManager and all mods to the latest versions and ran the game (under Windows) to make sure all was working properly.

Next I switched back to Linux, backed up and deleted my entire Sons game folder then copied the complete updated Windows version into my Linux Steam folder - and it worked immediately.

A million thanks for your help with this, great work, I am extremely grateful for your help.

Kind regards 🥇 👍 👎 💯

**Comment by SolacianAllie** [comment]
What about Red Mod Manager itsself?
i was unable to get it to run under linux

**Comment by hatesigningup** [comment]
> What about Red Mod Manager itsself? i was unable to get it to run under linux

Yeah, my reply above wasn't very accurate!  I should have said that the mods were working (...the "Red Loader" part), but NOT the actual RedModManager app itself.  Sorry for the confusion.  All of my installed mods work, apart from "UpdatesChecker"  - well it loads but it doesn't seem able to make an Internet connection.
